.devcontainer/Dockerfile

```diff
@@ -9,4 +9,22 @@ FROM mcr.microsoft.com/vscode/devcontainers/rust:0-${VARIANT}
 #     && apt-get -y install --no-install-recommends <your-package-list-here>
 
 ARG BINDLE_URL="https://github.com/deislabs/bindle/releases/download/v0.8.0/bindle-v0.8.0-linux-amd64.tar.gz"
-RUN curl -sL "$BINDLE_URL" | tar -xzf - -C /usr/local/bin bindle bindle-server
\ No newline at end of file
+RUN curl -sL "$BINDLE_URL" | tar -xzf - -C /usr/local/bin bindle bindle-server
+
+# Go installation, see https://go.dev/doc/install
+ARG GO_URL="https://go.dev/dl/go1.18.1.linux-amd64.tar.gz"
+RUN curl -sL "$GO_URL" | tar -xzf - -C /usr/local
+ENV PATH "$PATH:/usr/local/go/bin"
+
+# TinyGo installation, see https://tinygo.org/getting-started/install/linux/ for instructions
+ARG TINYGO_URL="https://github.com/tinygo-org/tinygo/releases/download/v0.22.0/tinygo_0.22.0_amd64.deb"
+RUN curl -sL "$TINYGO_URL" -o tinygo_amd64.deb \
+    && dpkg -i tinygo_amd64.deb \
+    && rm tinygo_amd64.deb
+
+# Install the gopls Go Language Server, see https://github.com/golang/tools/tree/master/gopls
+RUN go install golang.org/x/tools/gopls@latest
+
+# Install additional rust components (rust-analyzer)
+ARG RUST_ANALYZER_URL="https://github.com/rust-lang/rust-analyzer/releases/download/2022-04-11/rust-analyzer-x86_64-unknown-linux-gnu.gz"
+RUN curl -sL "$RUST_ANALYZER_URL" | gunzip -c - > /usr/local/bin/rust-analyzer && chmod +x /usr/local/bin/rust-analyzer
```

Comment:
I'd probably prefer to use `rustup component add rust-analyzer` here but it seems like `rust-analyzer` was added and then removed temporarily til it's stable, see https://github.com/rust-lang/rustup/pull/2560